### ISSUE TITLE
Quests display when overlays are active

### ISSUE THREAD

**Issue by martinrieder** [streetcomplete/StreetComplete#5762]
**Use case**

When editing overlay data in crowded spaces, the quests might be getting in the way of whatever you like to select. It would provide a big usability improvement if you could turn off quest display (as long as an overlay is active). 

**Proposed Solution**

~~Two different approaches that I see. Both could be implemented, since they are not conflicting.~~

1. Hide quests while overlay is selected and show them when "None" is selected. 
2. ~~Add a button next to the overlays to quickly select presets and create an empty one.~~

Note that option 1 implies that "None" becomes an overlay of its own, which could be renamed to "Quests" then. It would definitely be wise to add a setting for switching this behavior. It seems to me that this is the simpler option of the two.

~~Option 2 provides a lot more flexibility to the user. You might still want to see address related quests when you are in the Addresses overlay, etc.
To support that, the default list could contain additional presets that are related to these overlays.~~

EDIT: Option 2 removed due to https://github.com/streetcomplete/StreetComplete/issues/5762#issuecomment-2244487267

**Comment by riQQ** [comment]
See #3827 for your proposed option 2.

**Comment by mnalis** [comment]
**TL;DR:**  probably best to just wait if Maplibre max zoom increase will work around the issue. There are other more complex solutions too, but user behaviour change regarding using presets (or using SCEE functionality) might be a fix too.  

---

> 1. Hide quests while overlay is selected and show them when "None" is selected.

Note that if that were to be implemented, such behaviour would definitely would have to be **toggleable preference**[^1] IMHO. 

Here are few other possibilities:

3. perhaps Maplibre transition (https://github.com/streetcomplete/StreetComplete/pull/5693) would allow to zoom in more (i.e. https://github.com/streetcomplete/StreetComplete/issues/2878), thus sidestepping this issue (as you could zoom in to avoid accidentally clicking on too-close-by quest)
4. have an ability (but not the requirement!) to bind certain Overlay to certain preset; i.e. when you change to specific overlay, it would also change the preset if you set it up that way. But that seems like somewhat complex UI to set/clear such associations (more fitting perhaps for [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) fork, but that one already has _"quick change preset"_ button, so the point is moot)
5. in Overlay list, also have at a bottom (or top) of the list persistent _"Hide all quests"_ checkbox. That might perhaps be more usable (and certainly more discoverable) than option `1.`

That being said, perhaps when `3.` is completed the issue would be sidestepped without extra work (but then again, I've already been warned in the past that I'm being overly optimistic about Maplibre, so take that with huge grain of salt).

----

Another possible workarounds (not ideal, I know) might be:
- always select some low-density (of even "completely empty") preset before using overlays. 
  Yes, I know it is unergonomical, but it has improved from being [9-clicks-dance](https://github.com/streetcomplete/StreetComplete/issues/3827#issuecomment-1058071697) to now being **6 clicks** in StreetComplete (or **3 clicks** in SCEE); but you'd only have to do it once before starting using Overlays and once you no longer want to use Overlays at all)
- continue to use normal presets, but when using Overlay, first solve the quests that are in the way (or choose to `Uhh...` / `hide` them if you can't or won't answer them) **before** proceeding to click on the Overlay. When solved, they won't be in the way (and you'd improve the map at the same time)...
- if there are Quests which consistently get in your way and you rarely want to solve them, consider removing them from your main Preset for good (you can still keep them in your "Everything" preset for those moments when you feel bored). If they're "work" and not "fun", you're better off without them. Do not feel guilty about not solving them, the improved productivity and fun you're having will more than make up for any complex/boring quests you've skipped!

 
[^1]: E.g. I _always_ use Overlay in addition to quests (usually in less-crowded quest preset). Having to give up all Quests in order to be able to use single Overlay would be **huge** usability hit for me. (as instead having overlay enabled, I'd have to turn it off and back on after every few changes on the map in order to see/solve the Quests too. That would make Overlays practically unusable for me)

**Comment by westnordost** [comment]
Thank you, @mnalis for your thoughts, I agree with you.

Additionally, I would say that overlays are a power user feature, and power users tend to want to solve many quests. So, hiding all quests when an overlay is active must - if this was implemented - definitely be a setting, and it should not be enabled by default. But then again, there is not a big difference between selecting a preset with no quests active (or deselecting all quests, for people who don't really use different presets) and such a setting, just a few clicks more. So, introducing such a setting would also not be that useful.

On the idea number 5, this breaks a bit with the UI expectations. It is a popup menu where clicking on one menu item immediately selects that item and closes. Adding other elements to this menu would break that expectation. Putting more such options into the menu will thus lead to this menu being less and less a quick and easy way to change overlays. I.e. to not break UI consistency, if there were more options other than selecting another overlay, the overlay selection would probably need to be an own settings-like screen.

---

Regarding point 3 - MapLibre: Yes, a much higher zoom is possible in MapLibre.
